test/e2e-olm/kibana_test.go

```diff
@@ -0,0 +1,95 @@
+    "github.com/openshift/elasticsearch-operator/test/utils"
+    framework "github.com/operator-framework/operator-sdk/pkg/test"
+    "github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+    "k8s.io/apimachinery/pkg/types"
+)
+
+func TestKibana(t *testing.T) {
+    registerSchemes(t)
+    t.Run("kibana-group", func(t *testing.T) {
+        t.Run("KibanaDeployment", KibanaDeployment)
+    })
+}
+
+func KibanaDeployment(t *testing.T) {
+    ctx := framework.NewTestCtx(t)
+
+    namespace, err := ctx.GetNamespace()
+    if err != nil {
+        t.Fatal(err)
+    }
+    t.Logf("Found namespace: %v", namespace)
+
+    // get global framework variables
+    f := framework.Global
+    // wait for elasticsearch-operator to be ready
+    err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "elasticsearch-operator", 1, retryInterval, timeout)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    if err = createOpenshiftLoggingNS(f, ctx); err != nil {
+        t.Fatal(err)
+    }
+
+    if err = createElasticsearchSecret(f, ctx, constants.OpenshiftNS); err != nil {
+        t.Fatal(err)
+    }
+
+    uuid := utils.GenerateUUID()
+    esDeploymentName := fmt.Sprintf("elasticsearch-cdm-%v-1", uuid)
+    _, err = createElasticsearchCR(f, ctx, constants.OpenshiftNS, uuid)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    err = e2eutil.WaitForDeployment(t, f.KubeClient, constants.OpenshiftNS, esDeploymentName, 1, retryInterval, timeout)
+    if err != nil {
+        t.Errorf("timed out waiting for Deployment %v: %v", esDeploymentName, err)
+    }
+
+    if err = createTrustedCABundle(f, ctx); err != nil {
+        t.Fatal(err)
+    }
+    if err = createKibanaSecret(f, ctx); err != nil {
+        t.Fatal(err)
+    }
+
+    if err = createKibanaProxySecret(f, ctx); err != nil {
+        t.Fatal(err)
+    }
+    kibanaCR := createKibanaCR(namespace)
+    err = f.Client.Create(goctx.TODO(), kibanaCR, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+    if err != nil {
+        t.Errorf("could not create kibanaCR: %v", err)
+    }
+
+    err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, kibanaCRName, 1, retryInterval, timeout)
+    if err != nil {
+        t.Errorf("timed out waiting for Deployment kibana: %v", err)
+    }
+
+    appLogsConsoleLink := &consolev1.ConsoleLink{}
+    key := types.NamespacedName{Name: kibana.AppLogsConsoleLinkName, Namespace: namespace}
+    if err := f.Client.Get(goctx.TODO(), key, appLogsConsoleLink); err != nil {
+        t.Errorf("Console app logs link missing: %v", err)
+    }
+
+    infraLogsConsoleLink := &consolev1.ConsoleLink{}
+    key = types.NamespacedName{Name: kibana.InfraLogsConsoleLinkName, Namespace: namespace}
+    if err := f.Client.Get(goctx.TODO(), key, infraLogsConsoleLink); err != nil {
+        t.Errorf("Console infra logs link missing: %v", err)
+    }
```

Comment:
i wonder if we may be better off using something similar to what CLO does...
https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing-olm/test-010-deploy-via-olm-minimal.sh#L58

So, following that pattern we would add these to an assertions file:
https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing-olm/assertions

Currently we have to deploy ES for both tests, it might make sense to only have to do it once...

Reply:
First of all the console link assertion are Kibana-only related and furthermore I suggest to keep the same pattern here as we do i `elasticsearch_test.go` with ES nodes.

Second, the tests should stay independent and isolated on what they deploy/use. In addition kibana readiness require an ES deployment under `openshift-logging`. AFAIK, the NS-stickiness for Kibana affects multiple resources - ES, TrustedCABundle - and according to @igor-karpukhin it is intended.